### DISCUSSION TITLE
Revalidation using useFieldArray and YUP resolver

### DISCUSSION THREAD

**Discussion by patryk-fijalkowski** [react-hook-form/react-hook-form#8868] [General] [answered]
### Version Number

7.33.1

### Codesandbox/Expo snack

https://codesandbox.io/s/custom-hook-with-resolver-v7-forked-ezmkc2?file=/src/App.js
![image](https://user-images.githubusercontent.com/62008203/185309663-c628bfd5-cd58-424c-8099-a3265d917659.png)

### Steps to reproduce

1. Click on Submit
2. Type 2 characters in "test[0].firstName" input (min(3) validation triggered on change but is visible in errors object  but in different structure than required() validation)
3. Type 3 characters in "test[0].firstName" input (min(3) validation is valid but required() is still visible)
4. Clear input "test[1].firstName" and then type 3 characters.  (In the beggining you should see required() validation then min(3) and then field should be valid.
5. Press submit button. Now "test[0].firstName" is valid as well and works properly but "test[0].lastName" still has required() error and it doesn'r revalidate onChange. 

### Expected behaviour

"test[0].firstName" and "test[0].lastName" fields should work like "test[1].firstName" and "test[1].lastName" fields. After onSubmit click all fields should revalidate onChange mode as it is set in useForm config.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
`revaldiateMode` is default by onChange and it's working on my test in your codesandbox

https://user-images.githubusercontent.com/10513364/184995512-058c61db-cf24-4725-956a-d204557cffa5.mov

**Reply by patryk-fijalkowski** [reply]
Yep but "This field is required" still  is visible if you type more characters (more than min(3)). I don't understand why.

**Reply by bluebill1049** [reply]
check your validation.

**Reply by patryk-fijalkowski** [reply]
look at this field "test[0].lastName" here is only required validation.

**Reply by patryk-fijalkowski** [reply]
![image](https://user-images.githubusercontent.com/62008203/185069805-20a0dad4-c8fe-4824-b397-a65ca38b7438.png)
Look here I filled this two required inputs and it's still marked as invalid

**Comment by patryk-fijalkowski** [comment]
It looks like on revalidation RHF parse error structure returned by useYupValidationResolver. Every time resolver returns the same structure but in errors object in form state it looks like two separated section of errors.
![image](https://user-images.githubusercontent.com/62008203/185423288-43f820c1-af99-4ca0-890f-23e5fd98e111.png)

**Comment by patryk-fijalkowski** [comment, marked as answer]
![image](https://user-images.githubusercontent.com/62008203/185981635-54a59960-10a3-4cbc-bd49-7a7852ae9856.png)

I found the solution. It is neccessary to use resolver provided by @hookform/resolvers library.
